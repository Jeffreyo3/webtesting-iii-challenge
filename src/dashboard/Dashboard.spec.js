// Test away
import React from 'react';
import * as rtl from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import Dashboard from './Dashboard';


test ("<Dashboard /> snapshop", () =>  {

    const wrapper = rtl.render(<Dashboard />);

    expect(wrapper.asFragment()).toMatchSnapshot();

});

test("Initial state displays Unlocked and Open", () => {

    const { getByTestId } =  rtl.render(<Dashboard />);
    const lock = getByTestId('lock');
    const openClose = getByTestId('open-close');

    expect(lock.textContent).toBe('Unlocked');
    expect(openClose.textContent).toBe('Open');

})