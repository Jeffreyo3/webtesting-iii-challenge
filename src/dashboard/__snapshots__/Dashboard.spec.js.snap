// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> snapshop 1`] = `
<DocumentFragment>
  <div
    class="display panel"
  >
    <div
      class="led green-led"
      data-testid="lock"
    >
      Unlocked
    </div>
    <div
      class="led green-led"
      data-testid="open-close"
    >
      Open
    </div>
  </div>
  <div
    class="controls panel"
  >
    <button
      class="toggle-btn"
      data-testid="lockButton"
      disabled=""
    >
      Lock Gate
    </button>
    <button
      class="toggle-btn"
      data-testid="openButton"
    >
      Close Gate
    </button>
  </div>
</DocumentFragment>
`;
